contracts/examples/dao/DAO.sol

```diff
@@ -1,40 +1,60 @@
 pragma solidity ^0.5.10;
 
 import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
 import "./VentureEth.sol";
+import "../../drafts/voting/Voting.sol";
 
 
 /**
  * @title DAO
  * @dev The contract inherits from VentureEth.
  * @notice This is an exeprimental DAO (Decentralised Autonomous Organization) implementation. Use with caution.
  * 1. Issue the DAO tokens through an inital funding round.
- * 2. Propose a venture, and funding amount.
- * 3. Vote for a venture. Each DAO token is 1 vote. If venture is funded, tokens are reenabled for voting.
- * 4. Fund a venture. Votes must total 50% + 1 or more.
- * 5. Claim the venture tokens.
+ * 2. Propose a venture, and funding amount. You will have to stake the DAO's gage until either the venture is funded or you decide to drop the proposal.
+ * 3. Vote for a venture. Each DAO token is 1 vote. If venture is funded, or if you renounce your vote, you must manualy take back the tokens.
+ * 4. Fund a venture. Votes count must surpass the DAO's threshold for a majority.
+ * 5. Retrieve the venture tokens.
  * 6. Increase the DAO pool with returns (if any) on the tokens from a funded venture.
- * 7. Claim ether dividends from the DAO.
- * 8. Raise new capital for the DAO by restarting the funding round.
+ * 7. Claim ether dividends from the DAO on behalf of your DAO tokens.
  */
 contract DAO is VentureEth {
 
     using SafeMath for uint256;
+    using EnumerableSet for EnumerableSet.AddressSet;
 
-    mapping(address => uint256) public proposedFundingForVenture;
     uint256 public fundingPool;
-    mapping(address =>
-        mapping(address => uint256)
-    ) public votesForVentureByHolder;
-    mapping(address => uint256) public totalVotesForVenture;
-    mapping(address => uint256) public totalVotesByHolder;
-    mapping(address => address[]) public backersForVenture;
+    uint256 public gage;
```

Comment:
Let's simplify this by calling it `proposalFee` and making it non returnable. That would do to stop some of the trolling.